theano/tensor/opt.py

```diff
@@ -3421,7 +3423,26 @@ def local_fill_sink(node):
     c = node.op(*inputs)
     for model in models:
         c = T.fill(model, c)
-    return [c]
+
+    # The newly created node c doesn't has 'clients',
+    # so this iteration is took place with node.outputs[0]
+    replacements = {node.outputs[0]: c}
+    for client, cl_idx in node.outputs[0].clients:
+        if (hasattr(client, 'op') and
+                isinstance(client.op, T.Elemwise) and
+                not client.op == T.fill):
+            client_inputs = client.inputs[:]
+            client_inputs[cl_idx] = c
+            new_client = client.op(*client_inputs)
+
+            # Add clients to new_client
+            new_client.owner.outputs[0].clients = client.outputs[0].clients
+            r = local_fill_sink.transform(new_client.owner)
+            if r is False:
+                continue
+            replacements.pop(node.outputs[0], None)
```

Comment:
I just realized: it may be a bad thing to remove `node.outputs[0]` from `replacements` in general, because some clients may not be replaced, and in that case, they would still depend on the un-optimized `node.outputs[0]` instead of `c`.
Can you try to keep `node.outputs[0]` in `replacements`?
If there is a problem, let us know, and we will try to find a way to remove it only if *all* of the clients have been replaced.